lib/Sema/ConstraintGraph.cpp

```diff
@@ -343,9 +352,35 @@ void ConstraintGraphNode::retractFromInference(Constraint *constraint) {
   }
 }
 
-void ConstraintGraphNode::reintroduceToInference(Constraint *constraint) {
-  retractFromInference(constraint);
-  introduceToInference(constraint);
+void ConstraintGraphNode::retractFromInference(Type fixedType) {
```

Comment:
Looks like the logic in `retractFromInference(Type)` and `introduceToInference` could be unified into `a method i.e. `updateAfterBind` that would accept a callback that either introduces or retracts constraints...

Reply:
Good idea.